### PR TITLE
Update Elasticsearch transport client with high-level rest-client

### PR DESCRIPTION
**Pull request by Vathsan**
This PR fixes the issue by introducing java high-level rest-client to communicate with Elasticsearch thus removing the transport client.

[1] https://github.com/wso2/product-ei/issues/4069

### CODE REVIEW

File: components/org.wso2.ei.analytics.elk/src/main/java/org/wso2/ei/analytics/elk/publisher/ElasticStatisticsPublisher.java [hunk 1]
```diff
@@ -113,33 +115,24 @@ public static void process(PublishingFlow publishingFlow) {
...
-            log.error("No available Elasticsearch Nodes to connect. Please give correct configurations and" +
-                    " run Elasticsearch.", e);
-        } catch (ElasticsearchSecurityException e) {
-            log.error("Elasticsearch user lacks access to write.", e);
-            client.close();
+    public static void publish(List<String> jsonsToSend, RestHighLevelClient client) {
+
+        // Prepare and Send the bulk request
+        IndexRequest indexRequest = new IndexRequest("eidata");
+        for (String jsonString : jsonsToSend) {
```
Comment (sameeragunarathne) [inline components/org.wso2.ei.analytics.elk/src/main/java/org/wso2/ei/analytics/elk/publisher/ElasticStatisticsPublisher.java]: shouldn't we handle the null case for jsonsToSend array list?

File: components/org.wso2.ei.analytics.elk/src/main/java/org/wso2/ei/analytics/elk/publisher/ElasticStatisticsPublisher.java [hunk 2]
```diff
@@ -168,4 +161,31 @@ private static String getFormattedDate(long time) {
...
+        public void beforeBulk(long executionId, BulkRequest request) {
+            log.debug("Executing bulk [{" + executionId + "}]");
+        }
+
+        @Override
+        public void afterBulk(long executionId, BulkRequest request, BulkResponse response) {
+            if (response.hasFailures()) {
+                log.warn("Bulk [{" + executionId + "}] executed with failures");
+            } else {
+                log.debug("Bulk [{" + executionId + "}] completed in {" + response.getTook().getMillis() +
```
Comment (sameeragunarathne) [inline components/org.wso2.ei.analytics.elk/src/main/java/org/wso2/ei/analytics/elk/publisher/ElasticStatisticsPublisher.java]: add the log.isDebugEnabled check

File: components/org.wso2.ei.analytics.elk/src/main/java/org/wso2/ei/analytics/elk/observer/ElasticMediationFlowObserver.java [hunk 3]
```diff
@@ -273,41 +309,69 @@ private void getConfigurations() {
...
+        String trustStoreTypeInConfig = serverConf.getFirstProperty(ElasticObserverConstants.TRUST_STORE_TYPE_CONFIG);
+        String trustStorePasswordInConfig = serverConf.getFirstProperty(ElasticObserverConstants.TRUST_STORE_PASSWORD_CONFIG);
+        String sslEnabledInConfig = serverConf.getFirstProperty(ElasticObserverConstants.SSL_ENABLED_CONFIG);
 
         if (log.isDebugEnabled()) {
             log.debug("Configurations taken from carbon.xml.");
         }
 
-        int port = Integer.parseInt(portString);
+        //int port = Integer.parseInt(portString);
```
[the remark this answers is not in the document]
Comment (Vathsan) [reply components/org.wso2.ei.analytics.elk/src/main/java/org/wso2/ei/analytics/elk/observer/ElasticMediationFlowObserver.java]: Fixed in 835508f

File: components/org.wso2.ei.analytics.elk/src/main/java/org/wso2/ei/analytics/elk/publisher/ElasticStatisticsPublisher.java (hunk 1 above)
Comment (Vathsan) [reply components/org.wso2.ei.analytics.elk/src/main/java/org/wso2/ei/analytics/elk/publisher/ElasticStatisticsPublisher.java]: Added a null check in 835508f

File: components/org.wso2.ei.analytics.elk/src/main/java/org/wso2/ei/analytics/elk/publisher/ElasticStatisticsPublisher.java (hunk 2 above)
Comment (Vathsan) [reply components/org.wso2.ei.analytics.elk/src/main/java/org/wso2/ei/analytics/elk/publisher/ElasticStatisticsPublisher.java]: Fixed in 835508f

File: components/org.wso2.ei.analytics.elk/pom.xml [hunk 4]
```diff
@@ -35,38 +35,19 @@
...
         <dependency>
-            <groupId>org.elasticsearch.client</groupId>
-            <artifactId>x-pack-transport</artifactId>
-            <exclusions>
-                <exclusion>
-                    <artifactId>x-pack-core</artifactId>
-                    <groupId>org.elasticsearch.plugin</groupId>
-                </exclusion>
-            </exclusions>
+            <groupId>org.apache.httpcomponents</groupId>
```
Comment (milindaperera) [inline components/org.wso2.ei.analytics.elk/pom.xml]: move dependencies with versions to main root pom file under dependency management

File: components/org.wso2.ei.analytics.elk/pom.xml [hunk 5]
```diff
@@ -95,6 +76,11 @@
         </dependency>
     </dependencies>
 
+    <properties>
+        <elk.httpclient.version>4.5.8</elk.httpclient.version>
+        <elk.httpcore.version>4.4.11</elk.httpcore.version>
+    </properties>
```
Comment (milindaperera) [inline components/org.wso2.ei.analytics.elk/pom.xml]: move version properties to root pom

Comment (Vathsan) [reply components/org.wso2.ei.analytics.elk/pom.xml]: Fixed in a6b950256f312e59016a853e11d625a06ba07ae0

File: components/org.wso2.ei.analytics.elk/pom.xml (hunk 4 above)
Comment (Vathsan) [reply components/org.wso2.ei.analytics.elk/pom.xml]: As the elastic search client needs newer versions of httpcore and httpclient jar (we use older versions of these jars in parent pom) the dependencies are packed with the component itself.

**Pull request merged**